common/client/transaction_sender.go

```diff
@@ -0,0 +1,261 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"slices"
+	"sync"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink/v2/common/types"
+)
+
+type TransactionSender[TX any] interface {
+	SendTransaction(ctx context.Context, tx TX) (SendTxReturnCode, error)
+}
+
+// TxErrorClassifier - defines interface of a function that transforms raw RPC error into the SendTxReturnCode enum
+// (e.g. Successful, Fatal, Retryable, etc.)
+type TxErrorClassifier[TX any] func(tx TX, err error) SendTxReturnCode
+
+type sendTxResult struct {
+	Err        error
+	ResultCode SendTxReturnCode
+}
+
+const sendTxQuorum = 0.7
+
+// SendTxRPCClient - defines interface of an RPC used by TransactionSender to broadcast transaction
+type SendTxRPCClient[TX any] interface {
+	SendTransaction(ctx context.Context, tx TX) error
+}
+
+func NewTransactionSender[TX any, CHAIN_ID types.ID, RPC SendTxRPCClient[TX]](
+	lggr logger.Logger,
+	chainID CHAIN_ID,
+	chainFamily string,
+	multiNode *MultiNode[CHAIN_ID, RPC],
+	txErrorClassifier TxErrorClassifier[TX],
+	sendTxSoftTimeout time.Duration,
+) TransactionSender[TX] {
+	if sendTxSoftTimeout == 0 {
+		sendTxSoftTimeout = QueryTimeout / 2
+	}
+	return &transactionSender[TX, CHAIN_ID, RPC]{
+		chainID:           chainID,
+		chainFamily:       chainFamily,
+		lggr:              logger.Sugared(lggr).Named("TransactionSender").With("chainID", chainID.String()),
+		multiNode:         multiNode,
+		txErrorClassifier: txErrorClassifier,
+		sendTxSoftTimeout: sendTxSoftTimeout,
+	}
+}
+
+type transactionSender[TX any, CHAIN_ID types.ID, RPC SendTxRPCClient[TX]] struct {
+	services.StateMachine
+	chainID           CHAIN_ID
+	chainFamily       string
+	lggr              logger.SugaredLogger
+	multiNode         *MultiNode[CHAIN_ID, RPC]
+	txErrorClassifier TxErrorClassifier[TX]
+	sendTxSoftTimeout time.Duration // defines max waiting time from first response til responses evaluation
+
+	// TODO: add start/ stop methods. Start doesn't need to do much.
```

Comment:
I'm not sure that `Start` is needed as there is process of initialization or background job